### PR TITLE
KAA-1198 Raspberry Pi C SDK guide

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 1]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+**The further instructions must be executed on the host machine.**
+
+**Verified against:**
+
+ - **Host OS:** Ubuntu 14.04 LTS 32-bit LTS
+ - **Target OS:** Raspbian Jessie (2015-11-21)
+
+1. Download and install toolchain.
+
+        mkdir rpi_root && cd rpi_root
```
Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Please, use ``` for code with language tip.

Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: This is to avoid Jekyll's bug with code inside list

Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: LGTM in this case

**Comment by ohamov** [comment]
Because of the fact that encryption has been merged, the guide is no more valid. There is no need for `OpenSSL` build, hence the guide could be narrowed down to a few lines:
`mkdir rpi_root && cd rpi_root`
`git clone https://github.com/raspberrypi/tools.git`
`export PATH=$(pwd)/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian/bin:PATH`
_Create application_
`cmake -DKAA_MAX_LOG_LEVEL=3 -DCMAKE_TOOLCHAIN_FILE=../toolchains/rpi.cmake ..`
`make`

P.S. I think exporting the path is unnecessary. We can pass the compiler's name directly as a cmake flag.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 2]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+
+* TOC
+{:toc}
+
+The guide provides information on how to **cross-compile** C SDK for Raspberry PI.
+Alternatively, you can build the Kaa C endpoint SDK right on the Raspberry Pi board.
+Refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/) for further details.
+
+# Install dependecies
+
```
Comment (vadimol) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md:19]: dependencies

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 3]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+        mkdir rpi_root && cd rpi_root
+        export RPI_ROOT=$(pwd)
+
+        git clone https://github.com/raspberrypi/tools.git
+        export INSTALLDIR=$RPI_ROOT/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian
+        export PATH=$PATH:$INSTALLDIR/bin
+
+1. Download and install openssl.
+
```
Comment (vadimol) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md:32]: Guess that "2." must be here, however, taking to account Alexey's comment we can drop this at all.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 4]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+Alternatively, you can build the Kaa C endpoint SDK right on the Raspberry Pi board.
+Refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/) for further details.
+
+# Install dependecies
+
+**The further instructions must be executed on the host machine.**
+
+**Verified against:**
+
+ - **Host OS:** Ubuntu 14.04 LTS 32-bit LTS
```
Comment (vadimol) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: double LTS and probably 64 bit

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 5]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+ - **Target OS:** Raspbian Jessie (2015-11-21)
+
+1. Download and install toolchain.
+
+        mkdir rpi_root && cd rpi_root
+        export RPI_ROOT=$(pwd)
+
+        git clone https://github.com/raspberrypi/tools.git
+        export INSTALLDIR=$RPI_ROOT/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian
+        export PATH=$PATH:$INSTALLDIR/bin
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Prepend, don't append.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 6]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+
+1. Download and install openssl.
+
+        wget ftp://ftp.openssl.org/source/old/1.0.2/openssl-1.0.2f.tar.gz
+        tar -xf openssl-1.0.2f.tar.gz
+        cd openssl-1.0.2f && ./Configure --openssldir=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr shared os/compiler:arm-linux-gnueabihf-gcc
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Use `"` to protect the parameter from splitting.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 4 above)
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: @forGGe, you're Arch dude. I know you haven't tested it on Ubuntu.

I'd better leave the field empty and let QA fill it in.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 7]
```diff
@@ -5,4 +5,46 @@ permalink: /:path/
...
+# Install dependecies
+
+**The further instructions must be executed on the host machine.**
+
+**Verified against:**
+
+ - **Host OS:** Ubuntu 14.04 LTS 32-bit LTS
+ - **Target OS:** Raspbian Jessie (2015-11-21)
+
+1. Download and install toolchain.
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Dependencies are not installed. (`sudo apt-get install build-essentials cmake ...`.)

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 4 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Well, that's what http://docs.kaaproject.org/display/KAA/QNX+Neutrino+RTOS telling me :)

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 3 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md:32]: every `^[0-9]\.` will be converted to the proper list item by markdown processor automatically.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 4 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Added `TODO` here

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 5 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Done

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 6 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Removed this section according to @GamovCoder comments

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 2 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md:19]: Done

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 7 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Done

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 8]
```diff
@@ -5,4 +5,40 @@ permalink: /:path/
...
+Alternatively, you can build the Kaa C endpoint SDK right on the Raspberry Pi board.
+Refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/) for further details.
+
+# Install dependencies
+
+**The further instructions must be executed on the host machine.**
+
+**Verified against:**
+
+ - **Host OS:** **TODO [KAA-1198](http://jira.kaaproject.org/browse/KAA-1198): Verify agains Ubuntu 14.04 32-bit LTS**
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: `against`

Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: why 32 bit?

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 9]
```diff
@@ -5,4 +5,40 @@ permalink: /:path/
...
+
+1. Install build prerequisites
+
+        sudo apt-get install cmake build-essentials
+
+1. Install toolchain
+
+        mkdir rpi_root && cd rpi_root
+        git clone https://github.com/raspberrypi/tools.git
+        export PATH=$(pwd)/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian/bin:PATH
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: `PATH` -> `$PATH`

Comment (ohamov) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: For the 64-bit OS, should be used another compiler path, I think it'd be useful to be mentioned: `export PATH=$RPI_ROOT/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin:$PATH`

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 8 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Done

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md (hunk 9 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: Done

**Comment by vadimol** [comment]
Looks good for me

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md [hunk 10]
```diff
@@ -5,4 +5,40 @@ permalink: /:path/
...
+
+1. Install toolchain.
+
+        mkdir rpi_root && cd rpi_root
+        git clone https://github.com/raspberrypi/tools.git
+        export ARMLINUX_GCC=$(pwd)/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin/arm-linux-gnueabihf-gcc
+
+# Create application
+
+Now, dependencies are built and it is time to create Kaa application.
```
Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-RPi/index.md]: What dependencies have been built ?

**Comment by balldir** [comment]
LGTM

**Pull request merged**